python/mdvtools/stress_tests/test_json_write.py

```
@@ -0,0 +1,263 @@
+            save_func = save_json_atomic if use_atomic else save_json_old
+            try:
+                save_func(str(test_file), data)
+                return True
+            except Exception:
+                return False
+            
+        for i in range(iterations):
+            # Run multiple workers concurrently
+            threads = []
+            for j in range(5):  # 5 concurrent writers
+                thread = threading.Thread(target=lambda: worker(f"{i}-{j}"))
```

Ruff (0.8.2)

Function definition does not bind loop variable `i`

(B023)